### PR TITLE
restart from existing model added

### PR DESCRIPTION
**Pull request by LinjianMa**

### CODE REVIEW

File: cifar10-pytorch/tr_cg.py [hunk 1]
```diff
@@ -253,6 +266,7 @@ def main():
             [f'{props}', f'{train_loss}', f'{test_loss:4f}', f'{test_error}'])
         csv_file.flush()
 
+        torch.save(model.state_dict(), dirpath+"/saved_param")
```
Comment (ByzanTine) [inline cifar10-pytorch/tr_cg.py]: For the path, we may want to let the file indicate some hyper-params.
Like the name could be /model-bs=200-pros=50000 ??

File: cifar10-pytorch/tr_cg.py [hunk 2]
```diff
@@ -237,7 +246,11 @@ def main():
         **kwargs)
 
     model = Net().double().to(device)
-    model.zero_weights()
+    if (args.restart == False):
+        model.zero_weights()
```
Comment (ByzanTine) [inline cifar10-pytorch/tr_cg.py]: A more pythonic way would be 

if args.restart is False: 

or 

if not args.restart:

Comment (montaguegabe) [reply cifar10-pytorch/tr_cg.py]: Also maybe a name less ambiguous than "restart" like "load-model"

Comment (LinjianMa) [reply cifar10-pytorch/tr_cg.py]: Sure, I will modify them this afternoon

File: cifar10-pytorch/tr_cg.py (hunk 1 above)
Comment (montaguegabe) [reply cifar10-pytorch/tr_cg.py]: We will vary so many of the other parameters most likely that perhaps just the number of props is the most important one, and then we will change the file name once it is written?

File: cifar10-pytorch/results/model.csv [hunk 3]
```diff
@@ -0,0 +1,6 @@
+props train_loss test_loss test_accuracy
+710000 2.3025850929939833 2.222603 0.831
+890000 2.231902036900274 2.094803 0.7517
+1075000 2.098225116906218 2.022506 0.7237
+1345000 2.0311098891237718 2.016877 0.7105
+1605000 2.011537019706969 1.926151 0.6828000000000001
```
Comment (montaguegabe) [inline cifar10-pytorch/results/model.csv]: Sorry I committed my results files by accident

**Comment by montaguegabe** [comment]
Hang on model.csv isn't the only thing to delete

**Comment by montaguegabe** [comment]
I got it

File: cifar10-pytorch/tr_cg.py [hunk 4]
```diff
@@ -13,14 +13,25 @@
...
 
 logger = logging.getLogger('trcg')
-# TODO(yejiayu): Relax the filename.
-csv_file = open('train_test_data.csv', 'w', newline='')
 
+def get_file_prefix():
+    return 'cifar10_model'
+
+csv_path = join(results_dir, f'{get_file_prefix()}.csv')
+new_log = not Path(csv_path).exists()
```
Comment (ByzanTine) [inline cifar10-pytorch/tr_cg.py]: So if I understand correctly, this means if I have a csv already in results/cifar10_model, then nothing will be saved?

Comment (ByzanTine) [reply cifar10-pytorch/tr_cg.py]: Ic, you are trying to make it compatible with the saver...

Comment (LinjianMa) [reply cifar10-pytorch/tr_cg.py]: I think this is just used to check whether we need to write the header line in csv.

**Pull request merged**